internal/storage/memory.go

```diff
@@ -68,50 +89,67 @@ func (g *gauge) Store(key string, value float64) {
 }
 
 type MemStorage struct {
-	gauge   gauge
-	counter counter
+	gauge         gauge
+	counter       counter
+	filePath      string
+	storeInterval time.Duration
+	log           *zap.SugaredLogger
 }
 
-func NewMemStorage() MemStorage {
-	return MemStorage{gauge: gauge{g: make(map[string]float64)}, counter: counter{c: make(map[string]int64)}}
+func NewMemStorage(filePath string, storeInterval int, isRestore bool, log *zap.SugaredLogger) *MemStorage {
+	ms := MemStorage{
+		gauge:         gauge{g: make(map[string]float64)},
+		counter:       counter{c: make(map[string]int64)},
+		filePath:      filePath,
+		storeInterval: time.Duration(storeInterval) * time.Second,
+		log:           log,
+	}
+
+	if isRestore {
+		if err := ms.loadFromFile(filePath); err != nil {
+			ms.log.Errorf("during attempt to load from file, error occurred: %v", err)
+		}
+	}
+
+	if filePath != "" && storeInterval != 0 {
+		if err := ms.startStoring(); err != nil {
+			ms.log.Errorf("during initializing of new storage, error occurred: %v", err)
+		}
+	}
+
+	return &ms
 }
 
-func (storage *MemStorage) Store(metricType, name, value string) error {
+func (storage *MemStorage) Store(metricType, name string, value interface{}) error {
 	switch metricType {
 	case counterMetric:
-		v, err := strconv.ParseInt(value, 10, 64)
-		if err != nil {
-			return fmt.Errorf("failed to convert value %s to an integer: %w", value, err)
-		}
+		v := value.(int64)
```

Comment:
Необходимо проверить, что assertion был успешен

```suggestion
		v, ok := value.(int64)
```